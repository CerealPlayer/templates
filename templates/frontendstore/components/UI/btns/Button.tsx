import { ReactNode } from "react";

type props = { children: ReactNode; dis?: boolean };

export default function Button({ children, dis }: props) {
  return (
    <button
      disabled={dis}
      className="px-8 group py-2 relative flex justify-center items-center"
    >
      <div className="absolute top-0 left-0 w-full h-full border-b border-neutral-500 scale-x-0 group-hover:scale-x-100 transition-all duration-200 ease-out"></div>
      {children}
    </button>
  );
}
