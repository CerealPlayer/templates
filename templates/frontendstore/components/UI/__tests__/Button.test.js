import Button from "../Button";
import renderer from "react-test-renderer";

test("Button is disabled", () => {
  const component = renderer.create(<Button dis>Hey</Button>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
