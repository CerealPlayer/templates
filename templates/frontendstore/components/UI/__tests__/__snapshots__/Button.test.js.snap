// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button is disabled 1`] = `
<button
  className="px-8 group py-2 relative flex justify-center items-center"
  disabled={true}
>
  <div
    className="absolute top-0 left-0 w-full h-full border-b border-neutral-500 scale-x-0 group-hover:scale-x-100 transition-all duration-200 ease-out"
  />
  Hey
</button>
`;
